kpatch-build/create-diff-object.c

```diff
@@ -3332,6 +3325,13 @@ int main(int argc, char *argv[])
 	 */
 	kpatch_elf_teardown(kelf_patched);
 
+	/* create symbol lookup table */
+	base_locals = kpatch_elf_locals(kelf_base);
+	lookup = lookup_open(parent_symtab, mod_symvers, hint, base_locals);
+	free(base_locals);
+	kpatch_elf_teardown(kelf_base);
+	kpatch_elf_free(kelf_base);
```

Comment:
We should try to leave the kpatch_elf_teardown/free calls where they are, because they provide protection against accessing obsolete data later on, which was an occasional source of bugs in the past.

Would it work if we leave the teardown/free calls where they are, and put the kpatch_elf_locals() call immediately before them?  Then lookup_open() can still be done here, without having to also move the teardown/free calls.

Reply:
> Would it work if we leave the teardown/free calls where they are, and put the kpatch_elf_locals() call immediately before them? Then lookup_open() can still be done here, without having to also move the teardown/free calls.

We can leave the teardown where it is now, but I get segfaults in find_local_syms() unless kpatch_elf_free(kelf_base) is located after that call.

Reply:
Its this and hint as well. Updated the request.